packages/core-snapshots/src/codecs/codec.ts

```diff
@@ -0,0 +1,125 @@
+import { Models } from "@arkecosystem/core-database";
+import { Container } from "@arkecosystem/core-kernel";
+import { Blocks, Interfaces, Transactions, Utils } from "@arkecosystem/crypto";
+import { decode, encode } from "msgpack-lite";
+import { camelizeKeys } from "xcase";
+
+import { Codec as ICodec } from "../contracts";
+import { Codec as CodecException } from "../exceptions";
+
+@Container.injectable()
+export class Codec implements ICodec {
+    private static removePrefix(item: Record<string, any>, prefix: string): Record<string, any> {
+        const itemToReturn = {};
+
+        for (const key of Object.keys(item)) {
+            itemToReturn[key.replace(prefix, "")] = item[key];
+        }
+
+        return itemToReturn;
+    }
+
+    public blocksEncode(block: any) {
+        try {
+            const blockCamelized = camelizeKeys(Codec.removePrefix(block, "Block_"));
+
+            return Blocks.Serializer.serialize(blockCamelized, true);
+        } catch (e) {
+            throw new CodecException.BlockEncodeException(block.Block_id);
```

Comment:
Always include the original error object or error message when throwing a custom exception about something that involves data.